### ISSUE TITLE
channel_manager.getChannelParticipants() returns an empty list

### ISSUE THREAD

**Issue by LefterisJP** [raiden-network/raiden#892]
## Problem Definition

As can be seen from [this](https://github.com/raiden-network/raiden/pull/888#issuecomment-322793739) comment the channel manager's [`getChannelParticipants()`](https://github.com/raiden-network/raiden/blob/master/raiden/smart_contracts/ChannelManagerContract.sol#L37) returns an empty list even if there are currently open channels.

We have experienced this in the testnet with a channel manager for the "Raiden Test Token".

It can be verified by the following for a token on ropsten:

```js
var channel_manager_address = "0xff88b0bc757631a3de768bcb89d2c0664fc8b88d";
var channel_manager_abi = [{"constant":true,"inputs":[],"name":"getChannelsParticipants","outputs":[{"name":"","type":"address[]"}],"payable":false,"type":"function"},{"constant":true,"inputs":[{"name":"partner","type":"address"}],"name":"getChannelWith","outputs":[{"name":"","type":"address"}],"payable":false,"type":"function"},{"constant":true,"inputs":[],"name":"getChannelsAddresses","outputs":[{"name":"","type":"address[]"}],"payable":false,"type":"function"},{"constant":true,"inputs":[{"name":"node_address","type":"address"}],"name":"nettingContractsByAddress","outputs":[{"name":"","type":"address[]"}],"payable":false,"type":"function"},{"constant":true,"inputs":[],"name":"tokenAddress","outputs":[{"name":"","type":"address"}],"payable":false,"type":"function"},{"constant":false,"inputs":[{"name":"partner","type":"address"},{"name":"settle_timeout","type":"uint256"}],"name":"newChannel","outputs":[{"name":"","type":"address"}],"payable":false,"type":"function"},{"inputs":[{"name":"token_address","type":"address"}],"payable":false,"type":"constructor"},{"payable":false,"type":"fallback"},{"anonymous":false,"inputs":[{"indexed":false,"name":"netting_channel","type":"address"},{"indexed":false,"name":"participant1","type":"address"},{"indexed":false,"name":"participant2","type":"address"},{"indexed":false,"name":"settle_timeout","type":"uint256"}],"name":"ChannelNew","type":"event"},{"anonymous":false,"inputs":[{"indexed":false,"name":"caller_address","type":"address"},{"indexed":false,"name":"partner","type":"address"}],"name":"ChannelDeleted","type":"event"}];
var channel_manager = web3.eth.contract(channel_manager_abi).at(channel_manager_address);
```

and then just do  `channel_manager.getChannelsAddresses()` which will prove that there are a lot of channels registered for the token with the channel manager.

```js                                                                                                                                   
["0x51e207e928fc7dab63b10a2afbc79a072e7ce1de", "0x63d7a84f04525c840dc5f384f09eda8c305e59f1", "0x01eb720a36298fc5a04ed86cf864df28f78ec7e8", "0x128fd389d2b1576365025afab779a30c30928d7a", "0x8729f5d6a064ae4b926e051e2d0fcfdb65127271", "0xaa7ec7e21a3ab6e4532968f763efc82c0109863f", "0x13e3b39de924926e342572c0446cc544be910354", "0x0aba220ba13e59b514a896b281aadb26ba9b25cd", "0x46109bda432221d80a48609897bdc88f27f12dcb", "0xbab9d445fc3e937d148647885bd1218baba1915e", "0x92ddb169610c33bee446ccd848576b0915d41f1a", "0x5905855037a9b9e4480652a9de5332aa06e365fd", "0xb12ae461d356c3df572a2d9e34bc73fa6ae12ca3", "0x76fd7c1a14018fddb9c2117248ddd13f1668e300", "0x837ce9fff312342611f03804bd65ec5f0e8057f1", "0x3fd76c51cc7988716117f3636ab4db4a6a7c1b5d", "0x2f13edd5c0b1069cee3d4160752e50a8a9cc5cbd", "0xf6c19a08c61e9e4dae2cfa5f5dbec9245710888b", "0xff171018dcb2519a452fa507d45d6b52b9a966f0", "0x64c8b1a4a2fe018b352a93ee839d99aa94642709", "0x5419f95c5824088e4d6ce26b975ad788008e4acc", "0x818bc998e1f189234d130ca6f966b2e5a561b20b", "0x2bd8fcb4b749317419342966e99aa7a42d8ca074", "0x4a2782413fa6435efb8031a50dfc104a745a435e", "0xa0185ae8d470c290efb6ed83f5484a6f2411217a", "0xe665a6192c72b3c339f9843fd9f2d4e8e052663a", "0x86f5cc1577cb5cb814cd46afdea96e25a09fd31d", "0x066726932f3174f2e2abc7bca8a9fb8519051b74", "0x11a447c5407397eb99b13466ae07061a1b51e417", "0x55a2508eb83d4871ec31a4f23cc1642e9aaa2055", "0xf5890f45e1a063dcbd28e44aa4a25a084f2e688c", "0xee801d363372c7ea3b33d545142d9e6134b5fe96", "0xaf69bf1d6112bbbd5171f8d2d3be4b7e01d4bea4", "0xe31eff5ab0a8eb2086b9857430a8485797f32e37", "0xac69fbd8e7649dc2b4a6bde0ef62f9a37475be60", "0x2182a40db9da9a9eb396d43c05dfc59bf8b0cc22", "0xaa11c0f8d1ba8b04aed6604c51e16ea0f7e3c008", "0x807c3431e440c5172f724de26dd2fdca73cc234c", "0x82186f536a44201f56ef4d7777bf957b094defcb", "0x535635182b9ca66f3af75ed3983f43e222f8a19a", "0x7e84dd6537b2c4d92d4443d30200b8aa9c01d0d9", "0x2aac6b9d441e5d21e4ac36d29c497f004ce1d47d", "0x9456586830cc10cc766ba256634fa74bce3b9293", "0xc042d5c0e4064cfdbd8a96dc1446edbc93f5d224", "0x52a4dd4ae5397a0e9d83ff4ac04ecfd698bf7960", "0xddbcb4bd20d8747bd3638072724eac9aaea877f1", "0x1170301e90eda923e66383d535b843676d61cdc9", "0x98ab28afcfb627b2ccac8eb101989005091dcc5b", "0x0c4efcf509c458786345d524dbb023cf497d02c7", "0x65b9ddb4b643f1f7128ea99c1c99b2d5d2708b10", "0xca92d5c62ab2de4334f29b02afb6b10930e89f64", "0x05c782925b2e1e3383e7c97643ff7283c5dbb16e", "0x201b12738dc92c5530cb9d96ca4aa078f3b716e5", "0x077e98ed76c8ea7a51d3dbcfd9724f43a6cbed01", "0xb48417b5b29ea038f8ec45f4d04752cdb3870a25", "0xc80651df6adc101e1250710d1534bac272121146", "0x44a05420ae18db283d6bd8cba0ee092a92efa62a", "0x0f131b66121d6c29a802ea1eaaa6ec1385affa35", "0xc42c29b5d619d5db2ff8c965177ad1007ba61f7a", "0x4d1be3b53e5804e6b97d8dd96c9b6777c4d40afb", "0x93209cdc2b7d8b77706c4ba2890ec8dc94aa976d", "0x49e8afe85c0ac44f0f3c950728f2ca1fa902d2b3", "0xf56ddb341be09bd230717ee4c7b1603682cc99e5", "0x57fb499567ece90f43a911e73844ee213328ec76", "0x42f07e36134f11bc182b706ec8b6706a216a5af2", "0xc6bdd23cd2bca82eef4cf5b345acde2322d0e757", "0xe702ee556aa8df8db64a4d6e168c24f467caacd7", "0x5cae39c4fedf1fd069915d78e55aa86477eb59a5", "0x9e930556064adeb3ecb8b4e4bb7cc6fb832d48e1", "0x60ac5f1f236bd7d2c5d22fea1a33d55f0ee37498", "0x308ead619ca37ca4c0732de50cdd61229877568e", "0xd799012a585db47f8d63e7986a6b6833daf33da1", "0xf6e84185bcaa3026e65013dbd97d558a6f13a5d7", "0xca0776dedcb151daf171dab906823a84c26e204a", "0xfc71d9064672394b8437983a9f5cc9c338525469", "0xb9fe03e5a1659c68c5c54c244b009e6df0e5677c", "0x0965818960784be063746617c822dca95f14b0e1", "0x0e8097c67d1a16d863366b7f7cd9d4637c52a37d", "0x7145972a1920ebff39e3f4ccd4a1eecae62c7541", "0x9f7d2a3414c816b25651fe040b04ebd2fdee6aa3", "0x3d392c48268cd318b697670e06784794ac2804f4", "0x5d8ad4d42e33dde7576aac0b4671d7230b98d4c2", "0x947e7c7069f61acedb07fc85b9169f8ef15963a9", "0xcab2023c0921d72440ee2a7acfbbc97e0fbcd76c", "0xc9aa5038fc31d8cb24c55e9af95c1f7611a09fb7", "0xc360fb9ce630059dacb86254545f38329efb3adc", "0xb35be62cb604230804e94443b29864539be0700c", "0x2aed7fc7e82e284244107c340341bb7bad74f7fe", "0xdad44d6e990a0087df685e28e35a4fdd9dadcce8", "0xf543a13ec193bf1222ac36464b84131e0e5894f6", "0x4999990da9d66741e6656611d509fecec96a7aab", "0x57dd2a747d5a05d9e77a1b3e0353871508a42598", "0x404b6c779c2edbd310d0992fbf90522c4cef9391", "0x5022305a89335c9864510f128924557e15f67b1f", "0xb99009868ba025e204ced8ae979e22f2f6e8cf6b", "0x0cc5fa5a7a33aba83c30887518c73dce28f89251", "0x164732b1595bcb69cee5a4ec1d40a5556e274a11", "0x731519dd5c951d2442d595f0f28bd779e65d5b91", "0x17f259189ed387673e49de04940cc43d2d7b932c", "0xd67220fe0612d0e6a2ee9425b25b13dfa8c565f6", "0x7e1b513ba84643cb78af47eee40491b7db4befdb", "0x79f7fcfb04b58c57ee241d493a1bd63425d5031a", "0x3543dbaad2807777c0eb6c6469c7fc683d0097e9", "0x39f75db69c9a35fd895f7a97d75ff2e7e4a4acef", "0x3de50c09c384104ebb430f46ab666c011fb3fcef", "0xe7b147ca7d2c4d6b8c321f14686f5c36b391044c", "0x54053828e0456d21b6299b25da5141208b9a092b", "0xd4d58f186d00d01217e0f732830f878eb0087d71", "0xabfaab5416a32dc45511f5ea18bc872e7290b0d0", "0x738863d7a4a4a3712e57cbf801eb1eb5974c32ba", "0x59e5294f87232278ad7a90d953f26e19afce7bc9", "0x0a325662d2f4669ebff4ac891f1dd1c7568f7887", "0xadc644ece7fdc62af56eb75e636ef1a5b26be3b5", "0xc31cf44ae7ed7f6b1bb273b5cf734bfe57859cf0", "0xcc32a07bca1901013c86e03d0097cf1b8ad3abdb", "0x28594c3a4099a96a13e38f5415f2aac0606fa0e3", "0x358c8ff31de11a1f2535d3cd690bc63cea0d2438", "0xeb6db410c58b74e89fe6ea95211a042ab48b4446", "0x9509ce74cfee2cf071030de66fce49d4282456e3", "0x789024982e440bff5982953fe2165624251ffa9a", "0x2085252dc134769deb0f8102e690cb6b9ba4405d", "0x2bf3d664719a811e0b558c2b92232f95821d461f", "0xa490a268b61f5e1fb87582b96adfe9ad1bc485f9", "0xd5cc210f916fc928a2e947e52def71170015be31", "0xd37cd87d540c3168a485a65f555c788503eb2b32", "0xbab34c42fb04421d06501618cd39757b2ed5c7bd", "0xbc41405d5954b485dd2ef421a1726be76766cd87", "0x415cfd8f1cd3f94cceab11dfa26f1e6e033b5c90", "0x975311a041487ead5f08bd7fc0621de99351f85f", "0x0b6bbadbee300808a4c8a95768bb8fa96f91ed30"]
```
But then if you try to call `> channel_manager.getChannelsParticipants()` you get `[]`.

This, among other things, causes the steps described [here](https://github.com/raiden-network/raiden/pull/888#issuecomment-322757393) to fail and not properly instantiate a channel graph and as such we can't find a path to our direct channel partners.

## Tasklist

Figure out why. And if fixed then we can remove [this line](https://github.com/raiden-network/raiden/pull/888/files#diff-b62876f9087286c4355ccbdf6dc7b2baR250) added by #888.

**Comment by LefterisJP** [comment]
Another possible reason we see this in the testnet could be that the RTT token [channel manager](https://ropsten.etherscan.io/address/0xff88b0bc757631a3de768bcb89d2c0664fc8b88d) could have been deployed by an older version of Raiden before [this commit](https://github.com/raiden-network/raiden/commit/6d8a26f6ea89990b94523c63c4a37f780380cd84) from July 15 and as such the function we call could be different than what we think.

In order to make sure that this is not the reason we see the problem we should try to verify all contracts in the testnet.

**Comment by LefterisJP** [comment]
Regarding the [above](https://github.com/raiden-network/raiden/issues/892#issuecomment-323013545) comment we can be sure it's not an old contract. 

I verified all contracts deployed by @ulope [here](https://ropsten.etherscan.io/address/0x00ed9639e041d5014ce56d27015fabd07e9f4464). So now all of our currently deployed base contracts (NettingChannelLibrary, ChannelManagerLibrary, Registry and EndpointRegistry) in ropsten are verified.

I also verified the [channel manager](https://ropsten.etherscan.io/address/0xff88b0bc757631a3de768bcb89d2c0664fc8b88d#code) of the RTT contract and the problem can even by verified from etherscan's [read contract](https://ropsten.etherscan.io/address/0xff88b0bc757631a3de768bcb89d2c0664fc8b88d#readContract) feature.
